tools/mtf

```diff
@@ -27,16 +60,36 @@ while [[ $# -gt 0 ]]; do
       -l|--linters)
         AVOCADO_ARGS="$AVOCADO_ARGS $SITE_LIB/$MTF_TOOLS/*.py"
       ;;
+      --taglist)
+        print_tags
+        exit 0
+      ;;
+      -a|--action)
+        shift
+        ACTION="$1"
+      ;;
+      -t|--tags)
+        shift
+        TAGS="$TAGS --filter-by-tags=$1"
+      ;;
       *)
         AVOCADO_ARGS="$AVOCADO_ARGS $1"
       ;;
   esac
   shift
 done
 
+# used by mtf-log-parser
+test "$ACTION" = "run" && JSON_LOG=$(mktemp)
+test "$ACTION" = "run" && AVOCADO_ARGS="$AVOCADO_ARGS --json $JSON_LOG"
+
+# append filtering
+test -n "$TAGS"        && AVOCADO_ARGS="$AVOCADO_ARGS $TAGS"
+
 # preserve exist status, mtf command should finish same as testing.
-avocado run $AVOCADO_ARGS
+echo "RUNNING: $AVOCADO_CMD $ACTION $AVOCADO_ARGS"
+$AVOCADO_CMD "$ACTION" $AVOCADO_ARGS
 EXIT_STATUS=$?
-mtf-log-parser $JSON_LOG
```

Comment:
Why you removed mtf-log-parser? Any reason? Can you please paste how does it look like output now?

Reply:
it is not removed, just moved to another line, see code, because this log parser is intended just when you use default action `run` in case for example you will call `-a list` it leads to bad code